shadow.c

```diff
@@ -549,15 +559,29 @@ static char *get_full_path(const char *filename TSRMLS_DC)
 		SHADOW_G(curdir) = getcwd(NULL, 0);
 	}
 
+#if PHP_VERSION_ID < 50600
 	new_state.cwd = strdup(SHADOW_G(curdir));
+#else
+	new_state.cwd = estrdup(SHADOW_G(curdir));
```

Comment:
We can't use `estrdup/efree` on 5.4/5.5 because php code uses `realloc` on `new_state.cwd` and it fails with segfault. And vice versa, `strdup/free` doesn't work with 5.6 as it's uses `erealloc`. There's at least one affected usage of `realloc/erealloc` in `virtual_file_ex` function.

We could use `estrdup/efree` only and declare the extension >5.6 only, but I'm not sure about OD support.